### PR TITLE
Fix strategyType

### PR DESCRIPTION
**Pull request by black-adder**
The thrift0.9.2 sampling endpoint returns strategyType as an integer but thrift0.9.3 sampling endpoint returns strategyType as a string.

This fixes the remote sampler to use the string strategyType.

### CODE REVIEW

**Comment by black-adder** [comment]
We should be converting the JSON received from the agent to an internal thrift model like we do in go rather than doing string/integer comparisons like this. (I'll have to investigate if it's doable in node)

File: test/samplers/remote_sampler.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ describe('RemoteSampler', () => {
             remoteSampler._refreshSamplingStrategy();
         };
         server.addStrategy('service1', {
-            strategyType: 0,
+            strategyType: 'PROBABILISTIC',
```
Comment (oibe) [inline test/samplers/remote_sampler.js]: why not import these from remote_sampler.js?

File: test/samplers/remote_sampler.js [hunk 2]
```diff
@@ -139,7 +139,7 @@ describe('RemoteSampler', () => {
 
     it('should set per-operation sampler', (done) => {
         server.addStrategy('service1', {
-            strategyType: 0,
+            strategyType: 'PROBABILISTIC',
```
Comment (oibe) [inline test/samplers/remote_sampler.js]: Same as comment above.

File: test/samplers/remote_sampler.js (hunk 1 above)
Comment (yurishkuro) [inline test/samplers/remote_sampler.js]: Because it makes the test less strict

File: src/configuration.js [hunk 3]
```diff
@@ -155,16 +152,18 @@ export default class Configuration {
      *        as process-level tags on the Tracer itself.
      */
     static initTracer(config, options = {}) {
-        let reporters = [];
         let reporter;
         let sampler;
+        if (options.metrics) {
+            options.metrics = new Metrics(options.metrics);
```
Comment (yurishkuro) [inline src/configuration.js]: Isn't it possible that Metrics type is already passed in?

Comment (black-adder) [inline src/configuration.js]: hypothetically they can pass in whatever they want given it's javascript. We do have a comment above that says "metrics factory"

**Pull request merged**